src/crush/CrushWrapper.cc

```diff
@@ -3437,11 +3437,20 @@ int CrushWrapper::_choose_args_adjust_item_weight_in_bucket(
   }
   crush_choose_arg *carg = &cmap.args[bidx];
```

Comment:
@liewegas I've just noticed that you are trying to change the content of **cmap** without passing it in by reference or address?

Reply:
the crush_choose_arg_map that's passed by value is just a pointer:
<pre>
struct crush_choose_arg_map {
  struct crush_choose_arg *args; /*!< replacement for each bucket in the crushmap */
  __u32 size;                    /*!< size of the __args__ array */
};
</pre>
so it works, although function signature is a bit misleading.